app/src/androidTest/java/com/willowtree/vocable/room/MigrationTest.kt

```diff
@@ -61,4 +64,67 @@ class MigrationTest {
                 close()
             }
     }
+
+    @Test
+    @Throws(IOException::class)
+    fun migrate3to4() {
+        val categoryNameMap = HashMap<String, String>()
+        categoryNameMap["en_US"] = "My Sayings"
+
+        val phraseOneMap = HashMap<String, String>()
+        phraseOneMap["en_US"] = "Saying 1"
+
+        val phraseTwoMap = HashMap<String, String>()
+        phraseTwoMap["en_US"] = "Saying 2"
+
+        val mySayingsCategoryNameV3 = Converters.stringMapToJson(categoryNameMap)
+        val mySayingTestPhraseOneV3 = Converters.stringMapToJson(phraseOneMap)
+        val mySayingTestPhraseTwoV3 = Converters.stringMapToJson(phraseTwoMap)
+
+        val phraseOneId = UUID.randomUUID().toString()
+        val creationDateOne = System.currentTimeMillis()
+
+        val phraseTwoId = UUID.randomUUID().toString()
+        val creationDateTwo = System.currentTimeMillis()
+
+        helper.createDatabase(TEST_DB, 3).apply {
+            // Create mock My Sayings category
+            execSQL("INSERT INTO Category (category_id, creation_date, is_user_generated, localized_name, hidden, sort_order) VALUES ('${PresetCategories.USER_FAVORITES.id}', ${System.currentTimeMillis()}, 0, '$mySayingsCategoryNameV3', 0, ${PresetCategories.USER_FAVORITES.initialSortOrder})")
+
+            // Add custom phrases to My Sayings category
+            execSQL("INSERT INTO Phrase (phrase_id, creation_date, is_user_generated, last_spoken_date, localized_utterance, sort_order) VALUES ('$phraseOneId', $creationDateOne, 1, $creationDateOne, '$mySayingTestPhraseOneV3', 0)")
```

Comment:
It might be worth inserting a couple of phrases into shared prefs to test the part of the migration where there might be some leftover values from the 2 to 3 conversion.

Reply:
good idea!